src/context/cart-context.js

```diff
@@ -35,34 +40,59 @@ function CartProvider({children}) {
         catch(err) {
           console.log(err)
         }
+      }
     }
 
     const updateCartQuantity = async (token, productId, updateType) => {
-      try {
-        const res = await axios.post(`/api/user/cart/${productId}`, {
-          action: {
-            type: updateType,
-          }
-        }, {
-          headers: {
-            authorization: token
-          }
-        })
-        setMyCart(res.data.cart)
-      }
-    
-      catch(err) {
-        console.log(err)
+      if(productId !== null) {
```

Comment:
if product id is initialized than no need for null check you can just do if(productId)